test-data/unit/check-protocols.test

```diff
@@ -1940,6 +1940,17 @@ reveal_type(list(c for c in C))  # N: Revealed type is "builtins.list[__main__.C
 reveal_type(list(C))  # N: Revealed type is "builtins.list[__main__.C*]"
 [builtins fixtures/list.pyi]
 
+[case testIterableGetItemOnClass]
+class A:
```

Comment:
Let's also test:
- Metaclasses with `__getitem__` 
- `for` statements
- `list(iter(A()))` and `list(iter(B()))`